src/core/utils/power_calibration.hpp

```diff
@@ -0,0 +1,160 @@
+ * specified channel.
+ *
+ */
+class PowerCalibration : public InstanceLocator, private NonCopyable
+{
+public:
+    explicit PowerCalibration(Instance &aInstance);
+
+    /**
+     * Add a calibrated power of the specificed channel to the power calibration table.
+     *
+     * @param[in] aChannel                The radio channel.
+     * @param[in] aActualPower            The actual power in 0.01dBm.
+     * @param[in] aRawPowerSetting        A pointer to the raw power setting byte array.
+     * @param[in] aRawPowerSettingLength  The length of the @p aRawPowerSetting.
+     *
+     * @retval kErrorNone         Successfully added the calibrated power to the power calibration tabl.
+     * @retval kErrorNoBufs       No available entry in the power calibration table.
+     * @retval kErrorInvalidArgs  The @p aChannel, @aActualPower or @p aRawPowerSetting is invalid or the
+     *                            @ aActualPower already exists in the power calibration table;
+     *
+     */
+    Error AddCalibratedPower(uint8_t        aChannel,
+                             int16_t        aActualPower,
+                             const uint8_t *aRawPowerSetting,
+                             uint16_t       aRawPowerSettingLength);
+
+    /**
+     * Clear all calibrated powers from the power calibration table.
+     *
+     */
+    void ClearCalibratedPowers(void);
+
+    /**
+     * Set the target power for the given channel.
+     *
+     * @param[in]  aChannel      The radio channel.
+     * @param[in]  aTargetPower  The target power in 0.01dBm. Passing `INT16_MAX` will disable this channel.
+     *
+     * @retval  kErrorNone         Successfully set the target power.
+     * @retval  kErrorInvalidArgs  The @p aChannel or @p aTargetPower is invalid.
+     *
+     */
+    Error SetChannelTargetPower(uint8_t aChannel, int16_t aTargetPower);
+
+    /**
+     * Get the raw power setting for the given channel.
+     *
+     * Platform radio layer should parse the raw power setting based on the radio layer defined format and set the
+     * parameters of each radio hardware module.
+     *
+     * @param[in]      aChannel                The radio channel.
+     * @param[out]     aRawPowerSetting        A pointer to the raw power setting byte array.
+     * @param[in,out]  aRawPowerSettingLength  On input, a pointer to the size of @p aRawPowerSetting.
+     *                                         On output, a pointer to the length of the raw power setting data.
+     *
+     * @retval  kErrorNone         Successfully got the target power.
+     * @retval  kErrorInvalidArgs  The @p aChannel is invalid or @p aRawPowerSetting is nullptr.
+     * @retval  kErrorNotFound     The raw power setting for the @p aChannel was not found.
+     *
+     */
+    Error GetRawPowerSetting(uint8_t aChannel, uint8_t *aRawPowerSetting, uint16_t *aRawPowerSettingLength);
+
+private:
+    class CalibratedPower
```

Comment:
Maybe rename to `PowerSettingsEntry` (to indicate it is an entry in table/array)?
(just a suggestion, up to you if like to keep the original name it is all fine for me)

Reply:
Sure. Up to you. SGTM.